.github/workflows/integration-tests.yml

```diff
@@ -115,12 +104,50 @@ jobs:
       github_ci_changes: ${{ steps.ignore-filter.outputs.changes || steps.changes.outputs.github_ci_changes }}
       core_changes: ${{ steps.ignore-filter.outputs.changes || steps.changes.outputs.core_changes }}
       ccip_changes: ${{ steps.ignore-filter.outputs.changes || steps.changes.outputs.ccip_changes }}
+      should-run: ${{ (github.event_name != 'pull_request' && github.event_name != 'merge_group') || steps.filter.outputs.every_changed == 'true' || steps.filter.outputs.any_changed == 'true' }}
```

Comment:
Am I reading this correctly that any event other than PR or merge group will cause this to run?
I don't think this makes to do. We should enumerate the events that we expect explicitly to match in order to be read plainly (and without looking up docs) and to lock in a closed set that will not change. If we mean to match on push and scheduled runs, then we should say that as directly as possible.